regression-test/suites/insert_p0/txn_insert.groovy

```diff
@@ -328,19 +480,44 @@ suite("txn_insert") {
                 where txn_insert_dt4.id = txn_insert_dt2.id;
             """
             sql """
-                delete from txn_insert_dt2 where id = 1 or id = 5;
+                delete from txn_insert_dt2 where id = 1;
             """
             sql """
-                delete from txn_insert_dt5 partition(p_20000102) where id = 1 or id = 5;
+                delete from txn_insert_dt2 where id = 5;
+            """
```

Comment:
If we do insert and delete on the same table, then there is a problem that, if the delete is a predicate and implemented via push handler, then the delete take effect on the insert result. while if the delete is in `delete from` format then the delete does not take effect on the insert result.

It is ambiguous for users.

Reply:
BTW, delete sign also works as delete via push handler.

Reply:
If a dba changes the replication num of a table between two insert to a table, then what happens.

Reply:
Thread1 t2:      insert into table1
Thread2 t1: insert into table2 select table1, t3: insert into table2 select table1.

Is result of t2 visible to insert at t3?

Reply:
> Thread1 t2: insert into table1 Thread2 t1: insert into table2 select table1, t3: insert into table2 select table1.
> 
> Is result of t2 visible to insert at t3?

yes

Reply:
> If we do insert and delete on the same table, then there is a problem that, if the delete is a predicate and implemented via push handler, then the delete take effect on the insert result. while if the delete is in `delete from` format then the delete does not take effect on the insert result.
> 
> It is ambiguous for users.

For `delete from t1` and `insert t1` case, the delete does not work the next insert, see case 14.

Reply:
what if 
begin
insert into t1;
delete from t1;
insert into t2 select * from t1;
commit;

result of first insert is visible to delete but not visible to the second insert.

Reply:
> > Thread1 t2: insert into table1 Thread2 t1: insert into table2 select table1, t3: insert into table2 select table1.
> > Is result of t2 visible to insert at t3?
> 
> yes

For clarity,
Thread1 t2: begin  insert into table1; commit
thread2 t1:begin insert into table 2 select table2; t3: insert into table2 select table1; commit

If yes, what's our isolation level? ReadCommitted?